### PR TITLE
feat: RecentResults & RecentSearches added in analytics

### PR DESCRIPTION
**Pull request by anik-ghosh-au7**
- Issue Type - Enhancement

- Issue Link - https://www.notion.so/Arc-Dashboard-Add-analytics-views-for-recent-searches-results-e6fe6846ffb349848f11c5d8fe9ee7e1

- Description - Recent Searches and Recent Results components added for analytics view in arc-dashboard

### CODE REVIEW

**Review by bietkul — CHANGES_REQUESTED** [review]
@anik-ghosh-au7 Use Redux to manage state.

**Comment by anik-ghosh-au7** [comment]
@bietkul shifted the state management to redux

File: components/analytics/utils.js [hunk 1]
```diff
@@ -1016,3 +1016,79 @@ export const dateRangesColumn = (dates = dateRanges, columnItems = 4) =>
...
+		return defaults;
+	}
+	return defaults;
+};
+
+export const recentSearchesFull = (plan, displayReplaySearch) => {
+	if (plan !== 'growth' && plan !== 'bootstrap') {
+		return defaultColumns();
+	}
+	return [...defaultColumns(), ...(displayReplaySearch ? replaySearch : [])];
```
Comment (bietkul) [inline components/analytics/utils.js:1043]: @anik-ghosh-au7 Remove the logic based on plans.

File: components/analytics/utils.js [hunk 2]
```diff
@@ -1016,3 +1016,79 @@ export const dateRangesColumn = (dates = dateRanges, columnItems = 4) =>
...
+};
+
+export const recentResultsFull = (plan, ViewSource) => {
+	return [
+		...recentResultsCol('free'),
+		{
```
Comment (bietkul) [inline components/analytics/utils.js:1049]: @anik-ghosh-au7 Return all columns and remove the plan.

File: components/analytics/components/RecentSearches.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+		applyFilterParams({
+			filters,
+			filterId,
+			callback: this.fetchRecentSearches,
+			applyFilter: selectFilterValue,
+		});
+	}
+
+	componentDidUpdate(prevProps) {
+		const { filters } = this.props;
```
Comment (bietkul) [inline components/analytics/components/RecentSearches.js:30]: @anik-ghosh-au7 Error handling is missing. https://github.com/appbaseio/batteries/blob/b6259088c1f8c8d3de9456975c5a43f06cc40434/components/analytics/components/RequestDistribution/index.js#L57

File: components/analytics/components/RecentResults.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+		applyFilterParams({
+			filters,
+			filterId,
+			callback: this.fetchRecentResults,
+			applyFilter: selectFilterValue,
+		});
+	}
+
+	componentDidUpdate(prevProps) {
+		const { filters } = this.props;
```
Comment (bietkul) [inline components/analytics/components/RecentResults.js:30]: Error handling https://github.com/appbaseio/batteries/blob/b6259088c1f8c8d3de9456975c5a43f06cc40434/components/analytics/components/RequestDistribution/index.js#L57

**Review by bietkul — CHANGES_REQUESTED** [review]

File: modules/actions/analytics.js [hunk 5]
```diff
@@ -279,3 +281,59 @@ export function updateInsightStatus({ id, currentStatus, nextStatus }) {
...
+						{
+							appName,
+						},
+					),
+				),
+			)
+			.catch((error) =>
+				dispatch(
+					createAction(
+						AppConstants.APP.ANALYTICS.GET_POPULAR_SEARCHES_ERROR,
```
[the remark this answers is not in the document]
Comment (anik-ghosh-au7) [reply modules/actions/analytics.js]: @bietkul done

File: components/analytics/utils.js [hunk 6]
```diff
@@ -1016,3 +1016,56 @@ export const dateRangesColumn = (dates = dateRanges, columnItems = 4) =>
...
+			title: 'Source',
+			key: `rr-source${updateIndex()}`,
+			render: (item) => <ViewSource docID={item.key} index={item.index} />,
+		},
+	];
+};
+
+/**
+ * Get the no results seraches
```
Comment (bietkul) [inline components/analytics/utils.js:1045]: @anik-ghosh-au7 Fix comment

File: components/analytics/utils.js [hunk 7]
```diff
@@ -1016,3 +1016,56 @@ export const dateRangesColumn = (dates = dateRanges, columnItems = 4) =>
...
+	return doGet(
+		`${ACC_API}/_analytics/${getApp(appName)}recent-searches${getQueryParams({
+			size,
+			show_global: true,
+			...filters,
+		})}`,
+	);
+}
+/**
+ * Get the no results seraches
```
Comment (bietkul) [inline components/analytics/utils.js:1059]: @anik-ghosh-au7 Fix comment

**Review by bietkul — CHANGES_REQUESTED** [review]
@anik-ghosh-au7 Minor changes required

**Comment by anik-ghosh-au7** [comment]
demo - https://www.loom.com/share/89fa7e1a46dc455092d5837503ba50a3

**Pull request merged**